core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -43,21 +53,42 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 
 /**
  *
  */
 public class RangeAggregator extends BucketsAggregator {
 
-    public static class Range {
+    public static final ParseField RANGES_FIELD = new ParseField("ranges");
+    public static final ParseField KEYED_FIELD = new ParseField("keyed");
 
-        public String key;
-        public double from = Double.NEGATIVE_INFINITY;
-        String fromAsStr;
-        public double to = Double.POSITIVE_INFINITY;
-        String toAsStr;
+    public static class Range implements Writeable<Range>, ToXContent {
 
-        public Range(String key, double from, String fromAsStr, double to, String toAsStr) {
+        public static final Range PROTOTYPE = new Range(null, -1, null, -1, null);
+        public static final ParseField KEY_FIELD = new ParseField("key");
+        public static final ParseField FROM_FIELD = new ParseField("from");
+        public static final ParseField TO_FIELD = new ParseField("to");
+
+        protected String key;
+        protected double from = Double.NEGATIVE_INFINITY;
+        protected String fromAsStr;
+        protected double to = Double.POSITIVE_INFINITY;
+        protected String toAsStr;
+
+        public Range(String key, double from, double to) {
+            this.key = key;
+            this.from = from;
+            this.to = to;
+        }
+
+        public Range(String key, String from, String to) {
+            this.key = key;
+            this.fromAsStr = from;
+            this.toAsStr = to;
+        }
```

Comment:
can we make those two constructors call the 3rd one so that we can centralize validation (if we ever add some)